inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -703,3 +703,37 @@ def increase_restart_count_or_error(obj, eng):
         save_workflow(obj, eng)
     else:
         raise WorkflowsError('Workflow restarted too many times')
+
+
+def flatten_list(input_list):
+    if isinstance(input_list, (list, tuple)):
+        return [
+            element for innerList in input_list for element in flatten_list(innerList)
+        ]
+    return [input_list]
+
+
+@with_debug_logging
+def affiliations_for_hidden_collections(obj):
+    affiliations_mapping = current_app.config.get("AFFILIATIONS_TO_HIDDEN_COLLECTIONS_MAPPING", {})
+    affiliations = flatten_list(get_value(obj.data, 'authors.raw_affiliations.value', []))
+
+    affiliations_set = set()
+    for aff in affiliations:
+        affiliations_set.update(aff.upper().split())
```

Comment:
could you use a case-insensitive regex on word boundaries here instead? so things like `CERN,` work too.

Reply:
Why? I'm not sure if I understood you correctly as things like `CERN` works. IMHO regex here is unnecessary. I [updated test](https://github.com/inspirehep/inspire-next/pull/3987/files#diff-90a38869ac7ba833b44a3a7b5dfdc754fd7ad18c91f029c882e074150d171493R1454) for this case to be sure.

Reply:
I didn't write `CERN` but `CERN,` (with a trailing comma).

Reply:
Ah. You are right. :+1:  Fixed now.